message/producer_test.go

```diff
@@ -1,143 +1,94 @@
 package message
 
 import (
-	"errors"
+	"testing"
+	. "github.com/smartystreets/goconvey/convey"
+	mock "github.com/ONSdigital/dp-dimension-importer/message/message_test"
 	"github.com/ONSdigital/dp-dimension-importer/event"
-	"github.com/ONSdigital/dp-dimension-importer/message/message_test"
 	"github.com/ONSdigital/dp-dimension-importer/schema"
-	. "github.com/smartystreets/goconvey/convey"
-	"testing"
+	"github.com/ONSdigital/go-ns/log"
 	"time"
+	"errors"
 )
 
-func TestDimensionInsertedProducer_DimensionInserted(t *testing.T) {
-	Convey("Given a DimensionInsertProducer with valid configuration", t, func() {
-		outputChan := make(chan []byte)
-		closerChan := make(chan bool)
-		errorsChan := make(chan error)
-
-		insertedEvent := event.DimensionsInsertedEvent{
-			FileURL:    fileURL,
-			InstanceID: instanceID,
-		}
-
-		eventBytes, _ := schema.DimensionsInsertedSchema.Marshal(insertedEvent)
+var completedEvent = event.InstanceCompletedEvent{
+	InstanceID: "1234567890",
+	FileURL:    "/cmd/my.csv",
+}
 
-		marshallerMock := &message_test.MarshallerMock{
-			MarshalFunc: func(s interface{}) ([]byte, error) {
-				return eventBytes, nil
-			},
-		}
+func TestInstanceCompletedProducer_Completed(t *testing.T) {
+	Convey("Given InstanceCompletedProducer has been configured correctly", t, func() {
+		output := make(chan []byte)
```

Comment:
As mentioned consider using a buffered channel to remove the need for a separate go routine and the synchronisation. 

Here is an example of what I mean for this particular test: 

```

func TestInstanceCompletedProducer_Completed(t *testing.T) {
	Convey("Given InstanceCompletedProducer has been configured correctly", t, func() {
		output := make(chan []byte, 1)

		kafkaProducerMock := &mock.KafkaProducerMock{
			OutputFunc: func() chan []byte {
				return output
			},
		}
		marshallerMock := &mock.MarshallerMock{
			MarshalFunc: func(s interface{}) ([]byte, error) {
				return schema.InstanceCompletedSchema.Marshal(s)
			},
		}

		instanceCompletedProducer := NewInstanceCompletedProducer(kafkaProducerMock, marshallerMock)

		Convey("When given a valid instanceCompletedEvent", func() {

			err := instanceCompletedProducer.Completed(completedEvent)
			avroBytes := <-output

			Convey("Then no error is returned", func() {
				So(err, ShouldBeNil)

				Convey("And the exepcted bytes are sent to producer.output", func() {
					var actual event.InstanceCompletedEvent
					schema.InstanceCompletedSchema.Unmarshal(avroBytes, &actual)
					So(completedEvent, ShouldResemble, actual)
				})
			})
		})
	})
}
```

Reply:
I thought id fixed this - ill push up another version.